backend/pkg/web/handler/resource_fhir_test.go

```diff
@@ -0,0 +1,149 @@
+		Password: "test",
+	})
+	
+	//Pre adding the source data
+	w := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(w)
+	ctx.Set(pkg.ContextKeyTypeLogger, logrus.WithField("test", suite.T().Name()))
+	ctx.Set(pkg.ContextKeyTypeDatabase, suite.AppRepository)
+	ctx.Set(pkg.ContextKeyTypeConfig, suite.AppConfig)
+	ctx.Set(pkg.ContextKeyTypeEventBusServer, suite.AppEventBus)
+	ctx.Set(pkg.ContextKeyTypeAuthUsername, "test_user")
+
+	req, err := CreateManualSourceHttpRequestFromFile("testdata/Tania553_Harris789_545c2380-b77f-4919-ab5d-0f615f877250.json")
+	require.NoError(suite.T(), err)
+	ctx.Request = req
+
+	CreateManualSource(ctx)
+	//assert 
+	require.Equal(suite.T(), http.StatusOK, w.Code)	
+	type ResponseWrapper struct {
+		Success bool                    `json:"success"`
+		Source  models.SourceCredential `json:"source"`
+	}
+
+	var respWrapper ResponseWrapper
+	err = json.Unmarshal(w.Body.Bytes(), &respWrapper)
+	require.NoError(suite.T(), err)
+	
+	suite.SourceId = respWrapper.Source.ID
+}
+
+
+// AfterTest has a function to be executed right after the test finishes and receives the suite and test names as input
+func (suite *ResourcFhirHandlerTestSuite) AfterTest(suiteName, testName string) {
+	suite.MockCtrl.Finish()
+	os.Remove(suite.TestDatabase.Name())
+}
+
+func TestResourceHandlerTestSuite(t *testing.T) {
+	suite.Run(t, new(ResourcFhirHandlerTestSuite))
+}
+
+func (suite *ResourcFhirHandlerTestSuite) TestGetResourceFhirHandler() {
+	w := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(w)
+	ctx.Set(pkg.ContextKeyTypeLogger, logrus.WithField("test", suite.T().Name()))
+	ctx.Set(pkg.ContextKeyTypeDatabase, suite.AppRepository)
+	ctx.Set(pkg.ContextKeyTypeConfig, suite.AppConfig)
+	ctx.Set(pkg.ContextKeyTypeEventBusServer, suite.AppEventBus)
+	ctx.Set(pkg.ContextKeyTypeAuthUsername, "test_user")
+
+	ctx.AddParam("sourceId", suite.SourceId.String())
+	ctx.AddParam("resourceId", "57959813-8cd2-4e3c-8970-e4364b74980a")
+	
+	GetResourceFhir(ctx)
+
+	type ResponseWrapper struct {
+		Data *models.ResourceBase  `json:"data"`
+		Success bool               `json:"success"`
+	}
+	var respWrapper ResponseWrapper
+	err := json.Unmarshal(w.Body.Bytes(), &respWrapper)
+	require.NoError(suite.T(),err)
+	require.Equal(suite.T(),true,respWrapper.Success)
+	require.Equal(suite.T(),"Patient",respWrapper.Data.SourceResourceType)
+	require.Equal(suite.T(),suite.SourceId, respWrapper.Data.SourceID)
+	require.Equal(suite.T(),"57959813-8cd2-4e3c-8970-e4364b74980a", respWrapper.Data.SourceResourceID)
+	
+	// passing invalid sourceId & responseId
+	w.Body.Reset()
+	ctx.Params = []gin.Param {
+		{
+			Key: "sourceId", 
+			Value: "57959813-9999-4e3c-8970-e4364b74980a",
+		},
+		{
+			Key: "resourceId", 
+			Value: "57959813-9999-4e3c-8970-e4364b74980a",
+		},
+	}
+	
+	GetResourceFhir(ctx)
+
+	err = json.Unmarshal(w.Body.Bytes(), &respWrapper)
+	require.NoError(suite.T(),err)
+	require.Equal(suite.T(),false,respWrapper.Success)
```

Comment:
This should be moved into its own test. `TestGetResourceFhirHandler_WithInvalidSourceResourceId`